test/integration/test_mcp_oidc.py

```diff
@@ -415,19 +432,22 @@ def _start_mcp_server(
         yield f"{url}/mcp"
 
 
-@pytest.fixture(scope="session", params=[True, False])
+@pytest.fixture(scope="session", params=[False, True])
 def oidc_env(request, backend_aware_onprem_database_params) -> dict[str, str]:
     """
     The fixture builds a configuration for the `get_connection_factory`.
     Here we will explore two options:
-    1. The DB username is passed in a claim.
-    2. The users are anonymous and the MCP server has to use the default DB username.
-    """
-    env = {ENV_DSN: backend_aware_onprem_database_params["dsn"]}
+    1. The connection is made under the actual username.
+    2. The connection is made using pre-configured credentials, and the actual user
```

Comment:
May I suggest to also create four `env` scenarious:
- No `ENV_USER`, no `ENV_USERNAME_CLAIM`
- `ENV_USER`, no `ENV_USERNAME_CLAIM`
- no `ENV_USER`,  `ENV_USERNAME_CLAIM`
- `ENV_USER`,  `ENV_USERNAME_CLAIM`

Reply:
The first one should raise the ValueError. It is tested in unit tests. The second one in uninteresting here, it's like a normal user/password connection. The 3rd and 4th are implemented.